crypto/src/main/scala/io/iohk/ethereum/crypto/ECDSASignature.scala

```diff
@@ -64,6 +64,7 @@ object ECDSASignature {
       case Some(id) if v == negativePointSign => (id * 2 + newNegativePointSign).toByte
       case Some(id) if v == positivePointSign => (id * 2 + newPositivePointSign).toByte
       case None => v
+      case _    => throw new IllegalStateException("Cannot happen")
```

Comment:
Maybe set a val name for this case and add it to the exception message, just in case this happens we have more info available